nixos/modules/services/video/wivrn.nix

```diff
@@ -0,0 +1,170 @@
+{ config, pkgs, lib, ... }:
+let
+  inherit (lib) mkAliasOptionModule mkIf mkEnableOption mkPackageOption mkOption mkDefault optional optionalString optionalAttrs isDerivation getExe literalExpression maintainers;
+  cfg = config.services.wivrn;
+  configFormat = pkgs.formats.json { };
+
+  # For the application option to work with systemd PATH, we find the store binary path of
+  # the package, concat all of the following strings, and then update the application attribute.
+  # Application can either be a package or a list that has a package as the first element.
+  applicationExists = builtins.hasAttr "application" cfg.config.json;
+  applicationListNotEmpty = (
+    if builtins.isList cfg.config.json.application
+    then if builtins.length cfg.config.json.application == 0
+      then false
+      else true
+    else true
+  );
+  applicationCheck = applicationExists && applicationListNotEmpty;
+
+  applicationBinary = (
+    if builtins.isList cfg.config.json.application
+    then (builtins.head cfg.config.json.application)
+    else cfg.config.json.application
+  );
+  applicationStrings = builtins.tail cfg.config.json.application;
+
+  applicationPath = mkIf applicationCheck applicationBinary;
+
+  applicationConcat = (
+    if builtins.isList cfg.config.json.application
+    then builtins.concatStringsSep " " ([ (getExe applicationBinary) ] ++ applicationStrings)
+    else (getExe applicationBinary)
+  );
+  applicationUpdate = cfg.config.json // optionalAttrs applicationCheck { application = applicationConcat; };
+  configFile = configFormat.generate "config.json" applicationUpdate;
+in
+{
+  options = {
+    services.wivrn = {
+      enable = mkEnableOption "WiVRn, an OpenXR streaming application";
+
+      package = mkPackageOption pkgs "wivrn" { };
+
+      openFirewall = mkEnableOption "the default ports in the firewall for the WiVRn server";
```

Comment:
Shouldn't this be true by default?

Reply:
nope, we don't open firewalls by default.

Reply:
This appears to be true. Do you have a link to the documentation explaining why this is the case, because that doesn't make sense to me.